quantization/image_classification/migraphx/resnet50/e2e_migraphx_resnet_example.py

```diff
@@ -0,0 +1,388 @@
+    2. Download ILSVRC2012 validation dataset and development kit from http://www.image-net.org/challenges/LSVRC/2012/downloads.
+    3. Extract validation dataset JPEG files to 'ILSVRC2012/val'.
+    4. Extract development kit to 'ILSVRC2012/devkit'. Two files in the development kit are used, 'ILSVRC2012_validation_ground_truth.txt' and 'meta.mat'.
+    5. Download 'synset_words.txt' from https://github.com/HoldenCaulfieldRye/caffe/blob/master/data/ilsvrc12/synset_words.txt into 'ILSVRC2012/'.
+    
+    Please download Resnet50 model from ONNX model zoo https://github.com/onnx/models/blob/master/vision/classification/resnet/model/resnet50-v2-7.tar.gz
+    Untar the model into the workspace
+    '''
+
+    # Dataset settings
+    model_path = "./resnet50-v2-7.onnx"
+    ilsvrc2012_dataset_path = "/code/onnx_models/imagenet/ILSVRC2012"
+    #ilsvrc2012_dataset_path = "./ILSVRC2012"
+    augmented_model_path = "./augmented_model.onnx"
+    batch_size = 50
+    calibration_dataset_size = 1000  # Size of dataset for calibration
+
+    # INT8 calibration setting
+    calibration_table_generation_enable = True  # Enable/Disable INT8 calibration
+
+    # MIGraphX EP INT8 settings
+    os.environ["ORT_MIGRAPHX_FP16_ENABLE"] = "1"  # Enable FP16 precision
+    os.environ["ORT_MIGRAPHX_INT8_ENABLE"] = "1"  # Enable INT8 precision
+    os.environ["ORT_MIGRAPHX_INT8_CALIBRATION_TABLE_NAME"] = "calibration.flatbuffers"  # Calibration table name
+    os.environ["ORT_MIGRAPHX_ENGINE_CACHE_ENABLE"] = "1"  # Enable engine caching
+    execution_provider = ["MIGraphXExecutionProvider"]
+
+    # Convert static batch to dynamic batch
+    [new_model_path, input_name] = convert_model_batch_to_dynamic(model_path)
+
+    # Get calibration and prediction dataset size
+    [calibration_dataset_size, prediction_dataset_size] = get_dataset_size(ilsvrc2012_dataset_path,
+                                                                           calibration_dataset_size)
+
+    # Generate INT8 calibration table
+    if calibration_table_generation_enable:
+        print("Generating Calibration Table")
+        calibrator = create_calibrator(new_model_path, [], augmented_model_path=augmented_model_path)
+        calibrator.set_execution_providers(["ROCMExecutionProvider"])        
+        data_reader = ImageNetDataReader(ilsvrc2012_dataset_path,
+                                         start_index=0,
+                                         end_index=calibration_dataset_size,
+                                         stride=calibration_dataset_size,
+                                         batch_size=batch_size,
+                                         model_path=augmented_model_path,
+                                         input_name=input_name)
+        calibrator.collect_data(data_reader)
+        cal_tensors = calibrator.compute_data()
+
+        serial_cal_tensors = {}
+        for keys, values in cal_tensors.data.items():
+            #print(keys)
```

Comment:
Could you please delete these two comments?

Reply:
done